sdk/communication/communication-short-codes/test/public/manageUSProgramBriefs.spec.ts

```diff
@@ -27,80 +31,155 @@ describe(`ShortCodesClient - creates, gets, updates, lists, and deletes US Progr
+            pb,
+            expectedBriefMap[pb.id].brief,
+            "list all program briefs, byPage"
+          );
         }
       }
-      assert.isTrue(
-        foundTestProgramBrief,
-        "Test program brief was not returned in list of all program briefs"
-      );
+    }
+    // make sure all expected briefs were found
+    const allExpectedBriefsFound = Object.values(expectedBriefMap).every((expectedPB) => {
+      return expectedPB.found;
+    });
+    assert.isTrue(
+      allExpectedBriefsFound,
+      "Test program brief was not returned in list of all program briefs"
+    );
+    return totalNumberOfbriefs;
+  };
 
-      // delete program brief, ensure it was removed
-      const delRes = await client.deleteUSProgramBrief(uspb.id);
-      assert.isOk(delRes, "Deleting program brief failed");
-      assert.isFalse(
-        await doesProgramBriefExist(client, uspb.id),
-        "Delete program brief was unsuccessful, program brief is still returned"
-      );
+  const _testListUSProgramBriefs = async (
+    expectedProgramBriefs: USProgramBrief[]
+  ): Promise<boolean> => {
+    // list program briefs, validate test program brief is in the list
+    const totalNumberOfbriefs = await _listUSProgramBriefs(expectedProgramBriefs);
+
+    // test pagination, using 2 pages
+    const itemsPerPage = Math.floor(totalNumberOfbriefs / 2);
+    await _listUSProgramBriefs(expectedProgramBriefs, true, itemsPerPage);
+    return true;
+  };
+
+  it("can create, get, update, list, and delete a US Program Brief", async function () {
+    const testProgramBriefs = [getTestUSProgramBrief(), getTestUSProgramBrief()];
+    // override test brief id with variable id
+    const testProgramBriefIds = testProgramBriefs.map((pb, index) => {
+      const pbTestId = recorder.variable(`pb-var-${index}`, pb.id);
+      pb.id = pbTestId;
+      return pb.id;
+    });
+
+    await runTestCleaningLeftovers(testProgramBriefIds, client, async () => {
+      // validate upsert and update for each test brief
+      const testAndUpdateBrief = Object.values(testProgramBriefs).map(async (pb) => {
+        await _createTestProgramBrief(pb);
+        await _updateUSProgramBrief(pb);
+        return true;
+      });
+      assert.isOk(await Promise.all(testAndUpdateBrief));
+      // validate listProgramBriefs
+      assert.isOk(await _testListUSProgramBriefs(testProgramBriefs));
+
+      // delete program briefs, ensure it was removed
+      const testDeleteBrief = Object.values(testProgramBriefs).map(async (pb) => {
+        const delRes = await client.deleteUSProgramBrief(pb.id);
+        assert.isOk(delRes, "Deleting program brief failed");
+        assert.isFalse(
+          await doesProgramBriefExist(client, pb.id),
+          "Delete program brief was unsuccessful, program brief is still returned"
+        );
+        return true;
+      });
+      assert.isOk(await Promise.all(testDeleteBrief));
     });
-  }).timeout(50000);
+  }).timeout(80000);
```

Comment:
80 seconds for timeout? Is this necessary?

Reply:
leaving it again at 50000